docker/Dockerfile

```diff
@@ -1,44 +1,85 @@
+#
+# python:${PYTHON_VER}-slim --> (base) --> (install) --> (install-dev) -->  dev
+#                                 |            .              |
+#                                 V            .              V
+#                               final <.........          final-dev
+#
+# base (intermediate build target; basis for all other images herein)
+#   adds OS-level packaging dependencies for *running* Nautobot
+#   installs Pip and wheel
+#
+# install (intermediate build target)
+#   adds OS-level packaging dependencies for *installing* Nautobot, its dependencies, and its development dependencies
+#   installs Poetry
+#   uses Pip and Poetry to install Nautobot's Python dependencies (but *not* development dependencies)
+#   uses Poetry to build a Nautobot wheel
+#
+# install-dev (intermediate build target)
+#   uses Poetry to additionally install Nautobot's *development* dependencies
+#   adds files for Nautobot development server to run
+#
+# final-dev (development environment for Nautobot plugins)
+#   uses Pip to install Nautobot as a wheel
+#
+# dev (development environment for Nautobot core)
+#   uses Poetry to install Nautobot as editable
+#     (at runtime the requirement is to mount the live Nautobot source code to /source/; it won't run without that)
+#
+# final (production-ready environment):
+#   adds "nautobot" system user
+#   copies all installed Python packages, including Nautobot itself, from "install-nautobot"
+#   creates a self-signed SSL certificate
+#   sets up the system to run Nautobot in production with uwsgi
+#
+
+################################ Stage: base (intermediate build target; basis for all other images herein)
+
 FROM python:${PYTHON_VER}-slim as base
 
 ENV PYTHONUNBUFFERED=1 \
     NAUTOBOT_ROOT=/opt/nautobot \
     prometheus_multiproc_dir=/prom_cache
 
-# Install all OS package upgrades and dependencies
+# Install all OS package upgrades and dependencies needed to run Nautobot in production
 # hadolint ignore=DL3005,DL3008,DL3013
 RUN apt-get update && \
     apt-get upgrade -y && \
-    apt-get install --no-install-recommends -y git mime-support curl libxml2 libmariadb3 && \
+    apt-get install --no-install-recommends -y git mime-support curl libxml2 libmariadb3 openssl && \
     apt-get autoremove -y && \
     apt-get clean all && \
     rm -rf /var/lib/apt/lists/* && \
     pip --no-cache-dir install --upgrade pip wheel
 
 HEALTHCHECK --interval=5s --timeout=5s --start-period=5s --retries=1 CMD curl --fail http://localhost:8080/health/ || exit 1
 
-################################ Stage: dev
-# The dev container is used for development purposes but also to build the .whl for the final image
-# nautobot will be installed in editable mode in /source 
-FROM base as dev
+# Generate required dirs for later consumption
+RUN mkdir /opt/nautobot /prom_cache
+
+# Common entrypoint for all environments
+COPY docker/docker-entrypoint.sh /docker-entrypoint.sh
+ENTRYPOINT ["/docker-entrypoint.sh"]
```

Comment:
Moved these into the `base` image since they're needed for all three of `final`, `dev` and `final-dev`.